bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/discovery/DraytonWiserMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,90 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.draytonwiser.internal.discovery;
+
+import java.net.InetAddress;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.draytonwiser.internal.DraytonWiserBindingConstants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DraytonWiserMDNSDiscoveryParticipant} is responsible for discovering Drayton Wiser Heat Hubs. It uses the
+ * central MDNS Discovery Service.
+ *
+ * @author Andrew Schofield - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(service = MDNSDiscoveryParticipant.class, configurationPid = "mdnsdiscovery.draytonwiser")
+public class DraytonWiserMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(DraytonWiserMDNSDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(DraytonWiserBindingConstants.THING_TYPE_BRIDGE);
+    }
+
+    @Override
+    public String getServiceType() {
+        return "_http._tcp.local.";
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(final ServiceInfo service) {
+        if (service.getApplication().contains("http")) {
+            final ThingUID uid = getThingUID(service);
+
+            if (uid != null) {
+                logger.debug("Discovered Heat Hub '{}' with uid: {}", service.getName(), uid);
+                final Map<String, Object> properties = new HashMap<>(2);
+                final InetAddress[] addresses = service.getInetAddresses();
+
+                if (addresses.length > 0 && addresses[0] != null) {
+                    properties.put(DraytonWiserBindingConstants.ADDRESS, addresses[0].getHostAddress());
+                    properties.put(DraytonWiserBindingConstants.REFRESH_INTERVAL,
+                            DraytonWiserBindingConstants.DEFAULT_REFRESH_SECONDS);
+                }
+
+                return DiscoveryResultBuilder.create(uid).withProperties(properties)
+                        .withRepresentationProperty(uid.getId()).withLabel("Heat Hub - " + service.getName()).build();
```

Comment:
The representation property must be the name of the property.
```suggestion
                return DiscoveryResultBuilder.create(uid).withProperties(properties)
                        .withRepresentationProperty(DraytonWiserBindingConstants.ADDRESS).withLabel("Heat Hub - " + service.getName()).build();
```

Reply:
I see I haven't done the representation property correctly. In the thing xml it uses `serialNumber` as representation property. That value is set in the handler. So it should be serialNumber here. That should also be set as representation property of all things having a serial number. And also in the thing xml it should be added. The other representation property is room name and needs to be added to both code and thing xml

Reply:
Is this an issue with the refactor that you did? If so I'll leave you to make the necessary changes.

Reply:
yes probably :smile: I'll fix it. Do you know if the controller has a serial number or any identification?

Reply:
Sadly not, no. This is the json representation of the controller:
`{
            "id": 0,
            "NodeId": 0,
            "ProductType": "Controller",
            "ProductIdentifier": "Controller",
            "ActiveFirmwareVersion": "2.54.0",
            "ModelIdentifier": "WT714R1S0902",
            "DeviceLockEnabled": false,
            "DisplayedSignalStrength": "VeryGood",
            "ReceptionOfController": {
                "Rssi": -61
            },
            "ReceptionOfDevice": {
                "Rssi": 0,
                "Lqi": 255
            }
        }`

Unfortunately if you happen to have 2 heating systems, the json response would look identical.

Reply:
Resolved in af8d624